### PR TITLE
Feat/implement search bar mapcenter

### PR DESCRIPTION
**Pull request by DarinHajou**
PR summary: 

This pull request integrates a new search functionality into the Map Center, enabling users to search for coordinates such as locations, cities, and addresses. This enhancement makes it quicker and easier to find and set the map center directly.

Key changes:

* Enhanced Search Capability: The Map Center now supports searching by coordinates, city names, most landmarks, or addresses, facilitating precise and quick location finding.

* Updated Placeholder Text: The placeholder text within the Map Center's search bar has been updated to better reflect the new search capabilities.

* Temporary Text Fallback: Currently, the placeholder text for the general map search bar in map.tsx is set to a default fallback ("Search for sellers or items"). This is a temporary solution pending full integration of the new translation keys.

Just a note:

We're currently using the leaflet-control-geocoder plugin for our map center search functionality. While it covers most needs for MapCenter, we should review if we might need something more robust down the line.

### CODE REVIEW

**Comment by swoocn** [comment]
Good stuff @DarinHajou! ->

image

**Review by swoocn — APPROVED** [review]
Approved PR after some adjustments.  Please review the changes when you can and let me know your thoughts @DarinHajou.

File: src/components/shared/SearchBar/SearchBar.tsx
Comment (DarinHajou) [inline src/components/shared/SearchBar/SearchBar.tsx:1]: Neat improvement on the search component with the page prop addition. It streamlined the functionality across. Great job!

**Review by DarinHajou — COMMENTED** [review]
Neat improvement on the search component with the page prop addition. Great job!

**Pull request merged**